resources/automate.rb

```diff
@@ -0,0 +1,117 @@
+property :builder_pem, String, required: true
+property :platform, String
+property :platform_version, String
+
+load_current_value do
+  # node.run_state['chef-users'] ||= Mixlib::ShellOut.new('chef-server-ctl user-list').run_command.stdout
+  # node.run_state['chef-orgs'] ||= Mixlib::ShellOut.new('chef-server-ctl org-list').run_command.stdout
+  # current_value_does_not_exist! unless node.run_state['chef-orgs'].index(/^#{org}$/)
+end
+
+action :create do
+  # https://github.com/chef/delivery/issues/469
+  new_resource.config << "\ndelivery['chef_server_proxy'] = false" unless new_resource.config.include?('chef_server_proxy')
+
+  # Always make sure user and key provided to resource is at the bottom of the config, overriding duplicates.
+  new_resource.config << "\ndelivery['chef_username'] = '#{new_resource.chef_user}'"
+  new_resource.config << "\ndelivery['chef_private_key'] = '/etc/delivery/#{new_resource.chef_user}.pem'"
+
+  # Hardcode v1 runner search to automate-build-node
+  new_resource.config << "\ndelivery['default_search'] = 'tags:delivery-build-node'"
+
+  chef_ingredient 'automate' do
+    action :upgrade
+    channel new_resource.channel
+    version new_resource.version
+    config new_resource.config
+    accept_license new_resource.accept_license
+    platform new_resource.platform if new_resource.platform
+    platform_version new_resource.platform_version if new_resource.platform_version
+  end
+
+  directory '/etc/delivery'
+  directory '/etc/chef'
+
+  directory '/var/opt/delivery/license/' do
+    recursive true
+  end
+
+  {
+    '/var/opt/delivery/license/delivery.license' => new_resource.license,
+    "/etc/delivery/#{new_resource.chef_user}.pem" => new_resource.chef_user_pem,
+    '/etc/chef/validation.pem' => new_resource.validation_pem,
+    '/etc/delivery/builder_key' => new_resource.builder_pem,
+  }.each do |file, src|
+    chef_file file do
+      source src
+      user 'root'
+      group 'root'
+      mode '0600'
+    end
+  end
+
+  file '/etc/delivery/builder_key.pub' do
+    content lazy { "ssh-rsa #{[OpenSSL::PKey::RSA.new(::File.read('/etc/delivery/builder_key')).to_blob].pack('m0')}" }
+    user 'root'
+    group 'root'
+    mode '0644'
+  end
+
+  directory '/var/opt/delivery/nginx/etc/addon.d/' do
+    recursive true
+  end
+
+  file '/var/opt/delivery/nginx/etc/addon.d/99-installer_internal.conf' do
+    content <<-EOF
+      location /installer {
+        alias /opt/delivery/embedded/service/omnibus-ctl/installer;
+      }
+      EOF
+  end
+
+  ingredient_config 'automate' do
+    notifies :reconfigure, 'chef_ingredient[automate]', :immediately
+  end
+
+  if new_resource.enterprise.is_a?(String)
+    new_resource.enterprise = [new_resource.enterprise]
+    new_resource.enterprise.each do |ent|
+      execute "create enterprise #{ent}" do
+        command "delivery-ctl create-enterprise #{ent} --ssh-pub-key-file=/etc/delivery/builder_key.pub > /etc/delivery/#{ent}.creds"
```

Comment:
We could also do that in a future PR, but it is something we should do as all our documentation refers to it as `automate-ctl`